src/ragapp/admin-ui/sections/config/agent.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  AgentConfigSchema,
+  AgentConfigType,
+  DEFAULT_AGENT_CONFIG,
+  checkSupportedModel,
+  createAgent,
+  deleteAgent,
+  getAgents,
+  updateAgent,
+} from "@/client/agent";
+import { ExpandableSection } from "@/components/ui/custom/expandableSection";
+import { Tabs } from "@/components/ui/tabs";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2 } from "lucide-react";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { AgentTabContent } from "./agents/AgentTabContent";
+import { AgentTabList } from "./agents/AgentTabList";
+
+export const AgentConfig = () => {
+  const queryClient = useQueryClient();
+  const [activeAgent, setActiveAgent] = useState<string | null>(null);
+  const [agents, setAgents] = useState<AgentConfigType[]>([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const { data: fetchedAgents = [], isLoading: isLoadingAgents } = useQuery(
+    "agents",
+    getAgents,
+    {
+      onSuccess: (data) => {
+        // Sort agents by creation time
+        const sortedAgents = [...data].sort((a, b) => {
+          const dateA =
+            a.created_at instanceof Date
+              ? a.created_at
+              : new Date(a.created_at);
+          const dateB =
+            b.created_at instanceof Date
+              ? b.created_at
+              : new Date(b.created_at);
```

Comment:
can't we ensure in agent.ts that this is always a Date?

Reply:
Ah, i had the same question too. But still don't know the reason, just add these code for a safeguard because of a error in the UI 
```
TypeError: a.created_at.getTime is not a function
```

Reply:
i guess that's caused by the zod schema, this one doesn't work?

```
AgentConfigSchema = z.object({
  agent_id: z.string(),
  name: z.string(),
  role: z.string(),
  system_prompt: z.string(),
  tools: ToolsSchema,
  created_at: z.date(),
});
```